### ISSUE TITLE
Allow module usage

### ISSUE THREAD

**Issue by mattyclarkson** [cpplint/cpplint#29]
Due to global variables it is not possible to do something like:

``` py
import cpplint
cpplint.ProcessFile(filepathOne)
cpplint.ProcessFile(filepathTwo)
```

This would require refactoring all the global state into the `CpplintState` class and passing it between functions.

Would this be something that would be of interest to the project? We would like to use `cpplint` in our Python build tool and using it at a module would simplify things hugely.

**Comment by tkruse** [comment]
Since this would move the codebase away from upstream, closing as wontfix, but could be reopened in upstream.
